src/main/java/dk/sdu/se_f22/sharedlibrary/models/Product.java

```diff
@@ -1,83 +1,184 @@
     }
-    
-    @Override
-    public String toString(){ //overrides toString method, returning product name and price
-        return "Product: " + productAttributes.get(ProductAttribute.NAME) + " price: " + productAttributes.get(ProductAttribute.PRICE);
-    }
-    
-    public void print(){
-        System.out.println("Product : " + productAttributes.get(ProductAttribute.NAME)); //returns product name
-        
-        for(ProductAttribute pA : ProductAttribute.values()){ //runs through all product attributes
-            
-            if(pA == ProductAttribute.IN_STOCK){ //if product is in stock, prints the available locations
-                StringBuilder toPrint = new StringBuilder();
-                for(String s : getLocations()){
-                    toPrint.append(s).append("\t");
-                }
-                System.out.println("\t " + pA.alias + ": " + toPrint);
-                
-            }else {
-                
-                System.out.println("\t " + pA.alias + ": " + get(pA));
-            }
+
+    /**@param attributeName The name of the attribute to retrieve
+     * @return The value of the attribute referenced
+     * @throws IllegalArgumentException if the attribute input does not correspond to any of the attributes, whose value is a double
+     */
+    public double getDoubleValue(String attributeName){
+        return switch (attributeName) {
+            case ("price") -> this.getPrice();
+            case ("averageUserReview") -> this.getAverageUserReview();
+            case ("clockspeed") -> this.getClockspeed();
+            case ("weight") -> this.getWeight();
+            default -> throw new IllegalArgumentException(attributeName + "does not exist as a double attribute: ");
+        };
+    }
+
+    public UUID getUuid() {
+        return uuid;
+    }
+
+    public double getAverageUserReview() {
+        return averageUserReview;
+    }
+
+    public List<String> getInStock() {
+        return inStock;
+    }
+
+    public long getEan() {
+        return ean;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public Instant getPublishedDate() {
+        return publishedDate;
+    }
+
+    public Instant getExpirationDate() {
+        return expirationDate;
+    }
+
+    public String getCategory() {
+        return category;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getSize() {
+        return size;
+    }
+
+    public double getClockspeed() {
+        return clockspeed;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    public long getLongValue(String productAttribute) {
+        if(productAttribute.equals("ean")){
+            return this.getEan();
         }
+        throw new IllegalArgumentException(productAttribute + "does not exist as a double attribute: ");
+    }
```

Comment:
Same goes for this one, is it still necessary?

Reply:
will remove